### DISCUSSION TITLE
"Resolve type by string" feature

### DISCUSSION THREAD

**Discussion by innochenti** [BlackMATov/meta.hpp#30] [Q&A] [answered]
I like the idea of a library, but I think there is one feature that would improve its use: the ability to resolve a type by name. 
This would allow doing, for example, auto-scripting/auto-serialization of reflected objects. 
Something like:

`auto type = meta_hpp::get_type_by_name("rectangle");`

of course, it requires adding the name "rectangle" to 
`meta::class_<Rectangle>("rectangle");`

But that adds enormous flexibility to the library.

**Comment by BlackMATov** [comment, marked as answer]
So, we have similar functionality in this library. I call these scopes. You can think of it like regular C++ namespaces.

I've written a little example how to use it here: [Scopes](https://github.com/BlackMATov/meta.hpp/blob/48395d3b13f8ec473a32325e95f6f797d58594f9/manuals/meta_examples/scopes_example.cpp)

Scopes provide more advanced functionality than regular type naming. Also, this approach helps to avoid ambiguities of the same names in different namespaces or typedefs.

And, of course, you can do a for-each loop over all types in a scope and bind it to a scripting language or whatever.

In addition, you can add some free-form information about types, functions, parameters, etc. The library has the ability to do this using metadata functions and types. I don't have examples of this functionality right how, but you can find it in unit-tests.
